pkg/menu/menu.go

```diff
@@ -0,0 +1,72 @@
+package menu
+
+import (
+	"io"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+const menuWidth = 50
+
+// newParagraph will return a paragraph object with given initial text.
+func newParagraph(initText string, border bool, location int, wid int, ht int) *widgets.Paragraph {
+	p := widgets.NewParagraph()
+	p.Text = initText
+	p.Border = border
+	p.SetRect(0, location, wid, location+ht)
+	p.TextStyle.Fg = ui.ColorWhite
+	ui.Render(p)
+	return p
+}
+
+func processInput(input, warning *widgets.Paragraph, isValid func(string) (bool, string), uiEvents <-chan ui.Event) (string, error) {
+	// keep tracking all input from user
+	for {
+		e := <-uiEvents
+		if e.Type != ui.KeyboardEvent {
+			continue
+		}
+		switch e.ID {
+		// if user input control-d, raise error io.EOF meaning no more input.
+		case "<C-d>":
+			return "", io.EOF
+		case "<Enter>":
+			if isValid == nil {
```

Comment:
Passing nil pointers around can cause real trouble. 

A common pattern here would be to provide a func:
func alwaysValid(s string) (bool, string) {
return true, s
}

then for the cases where someone might have passed in nil, they can instead pass in alwaysValid
Then you can get rid of lines 35-37 because the function can always be called. 

Also I notice the isValid fund returns a bool and a string, and the processInput returns a string and an error. It is common for validators to return (string, bool) because that is similar to how maps work, e.g. if I have a map[string]string I say
x, ok := map["s"]
if ok {...}
and your isValid could be the same pattern:
isValid(string) (string, bool)
So lines 35-41 become this:
if s, ok := isValid(input.Text); ok {
return s, nil
}